torchci/lib/bot/retryBot.ts

```diff
@@ -1,6 +1,83 @@
 import { Probot } from "probot";
+import { getFlakyJobBeforeThisJob } from "../jobUtils";
 
+const SUCCESS_CONCLUSIONS = ["success"];
 const FAILURE_CONCLUSIONS = ["failure", "cancelled", "timed_out"];
+const ALLOWED_WORKFLOW_PREFIXES: { [key: string]: string[] } = {
+  pytorch: ["lint", "pull", "trunk", "linux-binary", "windows-binary"],
+  vision: [
+    "lint",
+    "Build Linux",
+    "Build Macos",
+    "Build M1",
+    "Tests on Linux",
+    "Tests on macOS",
+  ],
+};
+const DEFAULT_BRANCH = "main";
+const MAIN_BRANCH: { [key: string]: string } = {
+  pytorch: "master",
+  vision: DEFAULT_BRANCH,
+};
+
+async function retryPreviousWorkflow(
+  ctx: any,
+  owner: string,
+  repo: string,
+  branch: string,
+  workflowName: string,
+  jobs: any[]
+) {
+  let prevWorkflowId = 0;
+  // When a workflow finishes, we also want to go through all its successful jobs
+  // to check if the previous jobs are flaky with the following green (current),
+  // red (prev job), green (2-job before) pattern.  If this happens, retrying the
+  // previous red job
+  const retryJobs: number[] = [];
+  for (const job of jobs) {
```

Comment:
Use workflowId here to avoid looping through the list of job (save additional queries to Rockset)

Reply:
There is one query per workflow now, and it returns any flaky jobs from that workflow